worlds/ff6wc/__init__.py

```diff
@@ -339,8 +343,35 @@ def generate_early(self):
                 except ValueError as e:
                     raise ValueError(stesp_err_msg) from e
                 if stesp_min > 0 or stesp_max > 0:
-                    # TODO:
-                    raise NotImplementedError("starting espers is not supported")
+                    # pick a random number of starting espers between min & max specified
+                    num_start_espers = random.randint(stesp_min,stesp_max)
+                    # copy & shuffle the esper list
+                    mixed_espers = Rom.espers.copy()
+                    random.shuffle(mixed_espers)
+                    # pick each specific starting esper
+                    esper_index = 0
+                    while esper_index < num_start_espers:
+                        # add to the list of starting espers, 
+                        # this will be processed in create_items later
+                        self.starting_espers.append(mixed_espers[esper_index])
+                        # add the esper ID to the -sen flagstring
+                        sen_flag = sen_flag + str(Rom.espers.index(mixed_espers[esper_index]))
+                        # increment the counter
+                        esper_index = esper_index + 1
+                        # add a comma after the flagstring if this isn't the last one
+                        if esper_index < num_start_espers:
+                            sen_flag = sen_flag + ","
```

Comment:
just some tools in Python to do the same thing that you've done
- `random.sample` is for taking a number of things from a sequence
- `",".join(list_of_strings)` combines strings into one string separating them by commas
```suggestion
                    chosen_esper_indexes = self.random.sample(range(len(Rom.espers)), num_start_espers)
                    sen_flag = ",".join([str(id_) for id_ in chosen_esper_indexes])
                    self.starting_espers = [Rom.espers[i] for i in chosen_esper_indexes]
```